.gitignore

```diff
@@ -13,3 +13,8 @@ DerivedData
 **/gyb.pyc
 
 .*.sw[nop]
+
+CMakeFiles
+CMakeCache.txt
+cmake_*.cmake
```

Comment:
I'm not sure that there should be added.  We really should be discouraging in-tree builds, requiring out of tree builds.  I'd be okay with ignoring common output directories:

/out/
/.build/
/build/

But the listed files shouldn't be created.

If we are okay with using undocumented features, there is a way to prevent the in-tree builds IIRC.